### DISCUSSION TITLE
Crontab to restart some services?

### DISCUSSION THREAD

**Discussion by chrbar11** [roleoroleo/yi-hack-Allwinner-v2#798] [General]
Hello,

I'm using Yi Outdoor 1080p RFUS with firmware version 0.3.0 (h30ga) under base version 9.0.26.07.

I would like to test some commands via crontab (each hour) :
- to restart WiFi service (or disconnect and reconnect WiFi connection),
- to restart HTTPD service (Web Server),
- to restart the camera.

Could you please tell me which command lines do I have to use?

Thanks a lot for your help :)
Chris

**Comment by nemonoid** [comment]
You can write the crontab from the web interface (second tab) :

![Screenshot from 2024-01-04 00-56-36](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/142457275/ee1ff415-43cd-4a83-8666-f51b6c53f6f8)

I believe inserting the following should work to reboot the camera every hour :
`0 * * * * /sbin/reboot`

and I believe the following should reboot once a day :
`0 0 * * * /sbin/reboot`

( first you can connect with SSH and see if `/sbin/reboot` works to reboot the camera )

Regarding WiFi and HTTPD I don't know. But the YI boots up so fast that perhaps it's worth simply rebooting it..

**Comment by chrbar11** [comment]
Thanks @nemonoid for your help, command /sbin/reboot works to reboot the camera.

I meet some freeze randomly trouble with the camera model Yi Outdoor 1080p RFUS (no problem with Yi 1080p Home IFUS model on the same network).
Thus I would like to use command lines to restart WiFi and HTTPD services to do some tests and to workaround this trouble.

Does someone know command lines to restart WiFi and HTTPD services?
Thanks a lot for your help :)
Chris

**Reply by roleoroleo** [reply]
About httpd you should kill the process and restart it:
```
killall httpd
sleep 1
/tmp/sd/yi-hack/usr/sbin/httpd -p 80 -h /tmp/sd/yi-hack/www/ -c /tmp/httpd.conf
```
but if you want to use a more standard script, you should read the port from your config file (check the startup script system.sh).

About wifi I don't know exactly but you could try to kill wpa_supplicant and wait for the Yi processes to restart it.
Or there is a "strange" script /home/app/script/killwifi.sh wrote by Yi (never tested).

**Reply by chrbar11** [reply]
Thanks @roleoroleo for your help, I'll test that and post the results :)
